Sources/CouchbaseClusterManager/Config.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  Config.swift
+//  CouchbaseClusterManager
+//
+//  Created by Valerio Mazzeo on 09/11/2017.
+//  Copyright © 2017 Asensei Inc. All rights reserved.
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+//  See the GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+public struct Config {
+
+    public let cluster: Cluster
+
+    public let bucket: Bucket?
+
+    /// Time interval in between manager checks, in seconds.
+    public let interval: UInt32
+
+    public init(
+        cluster: Cluster,
+        bucket: Bucket?,
+        interval: UInt32
+    ) {
+        self.cluster = cluster
+        self.bucket = bucket
+        self.interval = interval
+    }
+
+    public init() throws {
+
+        var interval: UInt32 = 300
+
+        if let intervalString: String = ProcessInfo.processInfo.environment["CCM_INTERVAL"] {
+
+            guard let intervalUInt32: UInt32 = UInt32(intervalString) else {
+                throw Error.invalidKey("CCM_INTERVAL")
+            }
+
+            interval = intervalUInt32
+        }
+
+        self.init(
+            cluster: try Cluster(),
+            bucket: try? Bucket(),
+            interval: interval
+        )
+    }
+}
+
+public extension Config {
+
+    public struct Cluster {
+
+        /**
+         The cluster's name.
+
+         The manager will try to consolidate into the same cluster all the services which cluster property match this name.
```

Comment:
`The manager will try to consolidate all services that match the name into the same cluster`